lib/util/Components.js

```diff
@@ -114,8 +114,12 @@ Components.prototype.list = function() {
     }
     const id = this._getId(this._list[j].node);
     list[j] = this._list[j];
+    if (!list[j].usedPropTypes) {
+      list[j].usedPropTypes = [];
+    }
+
     if (usedPropTypes[id]) {
-      list[j].usedPropTypes = (list[j].usedPropTypes || []).concat(usedPropTypes[id]);
```

Comment:
Just one place actually: 
lib/rules/no-unused-prop-types.js:236
`
    function isPropUsed(node, prop) {
      for (let i = 0, l = node.usedPropTypes.length; i < l; i++) {
        const usedProp = node.usedPropTypes[i];
        if (
...
`